distribution/solr-distro/src/main/resources/server/solr/security.json

```diff
@@ -0,0 +1,24 @@
+{
+  "authentication": {
+    "blockUnknown": true,
+    "class": "solr.BasicAuthPlugin",
+    "credentials": {
+      "ddf": "aMp++h2DCQ8EbF+g1Rh6pRSRrDsU+WcUZk+TNHEJ4FA= 4PZmryDswuG+Vaau+sEYIkuqY+QpCINK7QxxOWy+yAk="
```

Comment:
❓ Should those be hardcoded or auto-generated?

Reply:
It is the hardcoded hash of the default password "changeit". The hardening guide will direct the user to change it to a password that complies with their organizational policies.

Reply:
There are a couple of directions to go in the future, depending on how much work we want to put into automation.

Option 1
 - On installation, generate a strong password. 
- Store the strong password as an encrypted config file entry
- Send an HTTP request to Solr to change the default password to the strong password.

Option 2
- Provide users with instruction on how to change the password in Solr 
- Let the user pick the password and update Solr
- The user enters the Solr password (typically through the Admin UI). The admin UI encrypts the password and stores it in a config file.

For either option, the HttpSolrClientFactory registers interest in the condif. When the password is updated, the HttpSolrClientFactory fetches it, decrypts it, and uses it when creating the HttpClient.

Reply:
PS: I updated DDF-4149 with the preceding comments

Reply:
This communication is internal only so I don't think we should involve the administrator. I am just worried whenever I see something hardcoded like this. I would rather we generate that automatically upon installation as we do for the node encryption key and store that in both the local Solr and us.

Reply:
I hear you.  It should be autogenerated. The system is capable of automating it. Autogenerating a password and using http to update the password in Solr is a 3-5 point ticket in itself.  I'm not able to squeeze it into this ticket.

@mcalcote Someone (maybe the blue team or maybe another team) should tackle DDF-4149 in sprint 47.

Reply:
I created a ticket for it and put it into a sprint for refinement 👍

Reply:
Definitely need to auto-generate a password upon install and we need hardening instructions in the meantime (that people won't follow). We also need to allow that password to be retrievable for an admin that needs to see it, in the event that they need to log directly into Solr.

Reply:
We will also need to make sure that this password (wherever it is stored) is exported as part of the migration:Export process.